src/client/action/list_databases.rs

```diff
@@ -0,0 +1,162 @@
+use std::{future::IntoFuture, marker::PhantomData};
+
+use bson::{Document, Bson};
+use futures_util::FutureExt;
+
+use crate::{Client, ClientSession, operation::list_databases as op, error::{Result, ErrorKind}, results::DatabaseSpecification, client::BoxFuture};
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+use crate::sync::Client as SyncClient;
+
+impl Client {
+    /// Gets information about each database present in the cluster the Client is connected to.
+    pub fn list_databases(
+        &self,
+    ) -> ListDatabases {
+        ListDatabases {
+            client: &self,
+            options: Default::default(),
+            session: None,
+            mode: PhantomData,
+        }
+    }
+
+    /// Gets the names of the databases present in the cluster the Client is connected to.
+    pub fn list_database_names(
+        &self,
+    ) -> ListDatabases<'_, Names> {
+        ListDatabases {
+            client: &self,
+            options: Default::default(),
+            session: None,
+            mode: PhantomData,
+        }
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl SyncClient {
+    /// Gets information about each database present in the cluster the Client is connected to.
+    pub fn list_databases(
+        &self,
+    ) -> ListDatabases {
+        self.async_client.list_databases()
```

Comment:
Since this is now literally just a pass-through (the runtime wrapping is handled by `ListDatabases::run`), it hints towards a tantalizing future where there's no need for a distinct `sync::Client` type.  That might not ever be possible, though, due to things like the async nature of construction and whatnot.